celery/concurrency/base.py

```diff
@@ -44,6 +45,9 @@ def apply_target(target, args=(), kwargs={}, callback=None,
     else:
         callback(ret)
 
+    # Switch to the tornado ioloop greenlet
+    future_executor.switch_to_ioloop()
```

Comment:
Yeah, recent works give me a idea about how to implement a more elegant interface between greenlet and tornado coroutine. I will start a daemon tornado ioloop thread and receive tornado coroutine in greenlet. When coroutine finishs, tornado ioloop can send result to related greenlet.